synapse/app/_base.py

```diff
@@ -70,6 +72,11 @@
 if TYPE_CHECKING:
     from synapse.server import HomeServer
 
+# Twisted injects the global reactor to make it easier to import, this confuses
+# mypy which thinks it is a module. Tell it that it a more proper type.
+reactor = cast(ISynapseReactor, _reactor)
```

Comment:
Unfortunate, but hopefully worth it to get the extra checking!

(If we end up making use of this elsewhere, perhaps we can do this import-cast dance in `synapse.types` or somewhere similar?)

Reply:
I had that thought briefly, but decided it wasn't worth tying things more tightly together at the moment!